ambry-store/src/main/java/com.github.ambry.store/IndexValue.java

```diff
@@ -62,18 +62,18 @@
 
   final static int INDEX_VALUE_SIZE_IN_BYTES_V0 =
       BLOB_SIZE_IN_BYTES + OFFSET_SIZE_IN_BYTES + FLAG_SIZE_IN_BYTES + EXPIRES_AT_MS_SIZE_IN_BYTES_V0
-          + ORIGINAL_MESSAGE_OFFSET_SIZE_IN_BYTES;
+          + RELATED_MESSAGE_OFFSET_SIZE_IN_BYTES;
 
   final static int INDEX_VALUE_SIZE_IN_BYTES_V1 =
       BLOB_SIZE_IN_BYTES + OFFSET_SIZE_IN_BYTES + FLAG_SIZE_IN_BYTES + EXPIRES_AT_SECS_SIZE_IN_BYTES_V1
-          + ORIGINAL_MESSAGE_OFFSET_SIZE_IN_BYTES + OPERATION_TIME_SECS_SIZE_IN_BYTES + ACCOUNT_ID_SIZE_IN_BYTES
+          + RELATED_MESSAGE_OFFSET_SIZE_IN_BYTES + OPERATION_TIME_SECS_SIZE_IN_BYTES + ACCOUNT_ID_SIZE_IN_BYTES
           + CONTAINER_ID_SIZE_IN_BYTES;
 
   private long size;
   private Offset offset;
   private byte flags;
-  private final long expiresAtMs;
-  private long originalMessageOffset;
+  private long expiresAtMs;
+  private long relatedMessageOffset;
```

Comment:
Maybe we can call it preUpdateMessageOffset?

Reply:
Any reason to prefer that name?

Reply:
`preUpdateMessageOffset` indicates the offset before this update record. `relatedMessageOffset` is somehow bidirectional.

Reply:
You are right - I changed this back to `originalMessageOffset` since it is always the offset of the "original" put message